validator/testcases/markup/markuptester.py

```diff
@@ -119,18 +116,15 @@ def _feed_parser(self, line):
         line = unicodehelper.decode(line)
 
         try:
-            self.feed(line + "\n")
-        except UnicodeDecodeError, exc_instance:
-            exc_class, val, traceback = sys.exc_info()
             try:
+                self.feed(line + "\n")
```

Comment:
This `try-catch` block catches an encoding error on the first try. After that, it force-converts it to unicode (does not try to catch unicode errors) and tries again. Either attempt, though, might raise another parser error, which is what the outer `try-catch` catches.